src/js/components/modals/InstallPackageModal.js

```diff
@@ -89,19 +92,29 @@ class InstallPackageModal
         currentTab: "defaultInstall",
         truncatedPreInstallNotes: true
       });
+
+      return;
     }
-    // If Opening
-    if (!props.open && nextProps.open) {
-      // If default - auto-install
-      if (!nextProps.advancedConfig) {
-        this.handleInstallPackage();
-      } else {
-        // Show advanced install
-        this.setState({
-          currentTab: "advancedInstall"
-        });
-      }
+
+    if (nextProps.isBetaPackage) {
+      this.setState({
+        betaTermsAccepted: false,
+        betaOptInProperties: BetaOptInUtil.getProperty(
+          nextProps.cosmosPackage.getConfig()
+        ),
+        currentTab: "betaTerms"
+      });
+
+      return;
     }
+
+    if (nextProps.advancedConfig) {
+      this.setState({ currentTab: "advancedInstall" });
+
+      return;
+    }
+
+    this.handleInstallPackage();
```

Comment:
This modal should be burned and it will be soon. It has a lot of anti-patterns and bad design including the use of `forceUpdate` which is a legacy part of our codebase. We need this method call here auto-install when a user clicks the `deploy` button.

Calling this already existed - I just refactored this method to untangle it a little.